resolver/custom_dns_resolver.go

```diff
@@ -14,33 +16,48 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type createAnswerFunc func(question dns.Question, ip net.IP, ttl uint32) (dns.RR, error)
+
 // CustomDNSResolver resolves passed domain name to ip address defined in domain-IP map
 type CustomDNSResolver struct {
 	configurable[*config.CustomDNS]
+	createAnswerFromQuestion createAnswerFunc
```

Comment:
Very minor but can you move this with the other data fields below, as opposed to the embedded types here? And do the same in `NewCustomDNSResolver`?